src/components/Table/ResourceTable.vue

```diff
@@ -9,41 +9,95 @@
       hide-details
     ></v-text-field>
   </v-card-title>
-  <v-data-table :headers="tableColumns" :items="resources" :search="search">
+  <v-data-table
+    :headers="tableColumns"
+    :items="filteredResources"
+    :search="search"
+  >
     <template v-slot:item.delete="{ item }">
-      <v-icon color="red" @click="onDelete(item.selectable.id)">mdi-delete</v-icon>
+      <v-icon color="red" @click="onDelete(item.selectable.id)"
+        >mdi-delete</v-icon
+      >
     </template>
     <template v-slot:item.edit="{ item }">
-      <router-link :to="{ name: 'Edit-Resource', params: { id: item.selectable.id } }">
-      <v-icon color="green">mdi-pencil</v-icon>
+      <router-link
+        :to="{ name: 'Edit-Resource', params: { id: item.selectable.id } }"
+      >
+        <v-icon color="green">mdi-pencil</v-icon>
+      </router-link>
+    </template>
+    <template v-slot:item.add="{ item }">
+      <router-link
+        :to="{ name: 'Add-Quantity', params: { id: item.selectable.id } }"
+      >
+        <v-icon color="blue">mdi-plus</v-icon>
       </router-link>
     </template>
   </v-data-table>
 </template>
 
 <script>
-import { ref, computed } from "vue";
+import { ref, computed, watch } from "vue";
 import { VDataTable } from "vuetify/labs/VDataTable";
 import { useStore } from "vuex";
 
 export default {
+  props: ["sortChoice"],
   components: {
     VDataTable,
   },
-  setup() {
+  setup(props) {
+    const internalSortChoice = ref(props.sortChoice);
+    watch(
+      () => props.sortChoice,
+      (newSortChoice) => {
+        internalSortChoice.value = newSortChoice;
+      }
+    );
     const store = useStore();
     const resources = computed(() => store.getters["resources/allResources"]);
-    const tableColumns = computed(() => store.getters["resources/getColumns"]);
+
+    const columnGettersMap = {
+      All: "resources/getColumns",
+      LinkingPart: "resources/getColumnsForLinkingPart",
+      Pearl: "resources/getColumnsForPearl",
+      Gemstone: "resources/getColumnsForGemstone",
+      PreciousMetal: "resources/getColumnsForPreciousMetal",
+    };
+
+    const selectedTableColumns = computed(() => {
+      const getterName = columnGettersMap[internalSortChoice.value];
+      if (getterName) {
+        return store.getters[getterName];
+      } else {
+        return store.getters["resources/getColumns"];
+      }
+    });
```

Comment:
Let's use a ternary here
```suggestion
      const getterName = columnGettersMap[internalSortChoice.value];
      return store.getters[getterName ? getterName : "resources/getColumns"]
    });
```